### ISSUE TITLE
%patch from not the start of the line does not work

### ISSUE THREAD

**Issue by mikhailnov** [rpm-software-management/rpm#1088]
Example piece of code in `%prep`

```
( cd asedriveiiie-usb
  %patch1 -p1
)
```

Result:
```
DEBUG: + cd asedriveiiie-usb
DEBUG: + %patch1 -p1
DEBUG: BUILDSTDERR: /var/tmp/rpm-tmp.A0WUoX: line 45: fg: no job control
```

But it works in rpm5. I had to put `%patch1` to the beginning of the line to make it work in rpm 4.15.1.

**Comment by pmatilai** [comment]
Rpm5 compatibility is not a goal, so whether something works there or not isn't relevant generally.

One could consider this a bug on its own merits though.

**Comment by mikhailnov** [comment]
Of course I understand that it is not a goal, I wanted to say that another implementation of rpm grammar behaves differently and at the same time I do not understand why it should behave like rpm4 does.

**Comment by dmnks** [comment]
One way to fix would be turning `%patch` into a proper macro, but I need to look closer.

**Comment by pmatilai** [comment]
Yep, but *that* is far easier said than done. We've been laying the foundations for that for a while now, see #2205 and various linked stuff.

**Comment by pmatilai** [comment]
This is got fixed by the way of #2205 in #2730.
